src/libcollections/binary_heap.rs

```diff
@@ -1194,8 +1194,8 @@ impl<T: Ord, I: IntoIterator<Item = T>> SpecExtend<I> for BinaryHeap<T> {
 }
 
 impl<T: Ord> SpecExtend<BinaryHeap<T>> for BinaryHeap<T> {
-    fn spec_extend(&mut self, ref mut other: BinaryHeap<T>) {
-        self.append(other);
+    fn spec_extend(&mut self, mut other: BinaryHeap<T>) {
```

Comment:
I can further clarify that yes, the BinaryHeap is consumed / moved into the method here. In the old code and new code. It's ultimately dictated by the Extend interface and not something that can be changed.